scripts/validate-agent-context.py

```
@@ -0,0 +1,187 @@
+        errors.append("owned_paths must stay relative to this worktree")
+    else:
+        owned = [Path(item) for item in data["owned_paths"]]
+        changed: set[str] = set()
+        commands = []
+        diff_base = "origin/master"
+        if isinstance(base_sha, str) and re.fullmatch(r"[0-9a-f]{40}", base_sha):
+            if subprocess.run(["git", "rev-parse", "--verify", f"{base_sha}^{{commit}}"], cwd=root, capture_output=True).returncode == 0:
+                diff_base = base_sha
+            else:
+                errors.append("base_sha does not resolve to a commit")
+        if subprocess.run(["git", "rev-parse", "--verify", f"{diff_base}^{{commit}}"], cwd=root, capture_output=True).returncode == 0:
```

Ruff (0.16.3)

[error] `subprocess` call: check for execution of untrusted input

(S603)